initial jt cal script

aegger13 opened the pull request:
Initial jt calibration script to run through ARP

ARP/SLURM output:
2020-06-05 17:28:32,784 - DEBUG - slurm_script:calc_intensity_bounds - Getting intensity data for {i_mon}
2020-06-05 17:28:32,860 - DEBUG - slurm_script:calc_intensity_bounds - Lower:Upper bounds for intensity monitor: 1.97:3.06
2020-06-05 17:28:32,863 - DEBUG - slurm_script:det_azav - Getting azimuthal values for detector: DsaCsPad
2020-06-05 17:28:34,132 - DEBUG - slurm_script:det_azav - Found average azimuthal average array from all events
2020-06-05 17:28:34,133 - DEBUG - slurm_script:calc_peak_and_intensity - Fitting Guassian of average azimuthal binned array
2020-06-05 17:28:34,144 - DEBUG - slurm_script:calc_peak_and_intensity - Peak found at 147, with and intensity of 237.04
2020-06-05 17:28:34,146 - DEBUG - slurm_script:write_file - Writing results to /reg/d/psdm/cxi/cxilr6716/calib/

I put the report.html file in same location.  Just review for now, I'll put proper location after talking to Murali.

Here is report generated.
![Screen Shot 2020-06-12 at 11 45 17 AM](https://user-images.githubusercontent.com/15186068/84536225-390c2c80-aca2-11ea-911a-8e092aa16c8b.png)

ZryletTC commented on arp_scripts/submit_jt_cal.sh:
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+source /reg/g/pcds/engineering_tools/xpp/scripts/pcds_conda
```
Why xpp?

silkenelson replied:
Historical: I've defaulted to XPP as I had control over where that pointed and I did the smalldata_tools stuff. It should probably use "expname[0:3]" instead of xpp or point to latest_release (whatever that directory was called again).

aegger13 commented:
Hi @silkenelson,
I removed references to q
I'm using np.std now, I was testing something and left that other code in there for some reason
I moved calc_intensity_bounds outside of class.  I'm using a class to organize my data and properties/variables, once we like what's happening and understand all the pieces we can change to a module with some useful utility

Something I noticed is that we are not seeing an image on the detector when we have something on intensity monitor (it's using gdet) so I'm sure it will be nicer when we have wave8 or something closer to the action.  Let me know how you want to deal with that, maybe a second filtering on the azav data?

When you talk about peak bin max #, you just want the maximum intensity value at the peak location in the azav array?  I'll add that, just want to make sure that's what you are looking for.  I didn't do anything with min as it will be basically zero with the dropout data included, but will do this when we know how to deal with 'dark' shots

I added the plot you requested for i0 vs peak and a ratio line, check it out, not sure if that's what you were looking for

For the distribution around 'golden' value, is it just a histogram with a vertical line showing the azav/i0 ratio with all azav/i0 ratios for other events?

Anyway, it's a next iteration, so let's just keep doing this until it seems good :)

aegger13 commented:
I'm going to merge this as it's getting stale and I need this code in master to work with other branches.  This will be improved as we start to test in real scenarios.
